projects/lib/src/board/board.cpp

```diff
@@ -529,14 +534,23 @@ bool Board::setFenString(const QString& fen)
 		if (square >= boardSize)
 			return false;
 
-		pieceStr.append(c);
-		Piece piece = pieceFromSymbol(pieceStr);
-		if (!piece.isValid())
-			continue;
+		// read ahead for multi-character symbols
+		int l, lsym_minus1 = 0;
+		for (l = 0; l < maxsymlen && i + l < token->length(); l++)
```

Comment:
Your feedback yields the necessary fresh ideas. It is strange how I sometimes do not make or forget about obvious improvements.